sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1628,6 +1642,46 @@ static bool setEnvVar(const char *name, const char *value) {
   return true;
 }
 
+static class ZeUSMImport {
+  // Pointers to functions that import/release host memory into USM
+  ze_result_t (*zexDriverImportExternalPointer)(ze_driver_handle_t hDriver,
+                                                void *, size_t);
+  ze_result_t (*zexDriverReleaseImportedPointer)(ze_driver_handle_t, void *);
+  bool ImportSupported;
+
+public:
+  ZeUSMImport() : ImportSupported{false} {}
+  void setZeUSMImport(pi_platform Platform) {
+    // Check if USM hostptr import feature is available.
+    ze_driver_handle_t driverHandle = Platform->ZeDriver;
+    bool ImportSupported =
+        ZE_CALL_NOCHECK(
+            zeDriverGetExtensionFunctionAddress,
+            (driverHandle, "zexDriverImportExternalPointer",
+             reinterpret_cast<void **>(&zexDriverImportExternalPointer))) == 0;
+    if (ImportSupported) {
+      ZE_CALL_NOCHECK(
+          zeDriverGetExtensionFunctionAddress,
+          (driverHandle, "zexDriverReleaseImportedPointer",
+           reinterpret_cast<void **>(&zexDriverReleaseImportedPointer)));
+      // Turn on hostptr import only if supported and explicitly requested.
+      USMHostPtrImportEnabled = true;
+      // Hostptr import is only possible if piMemBufferCreate receives a
+      // hostptr as an argument. The SYCL runtime does that only when
+      // SYCL_HOST_UNIFIED_MEMORY is enabled. Therefore we turn it on.
+      setEnvVar("SYCL_HOST_UNIFIED_MEMORY", "1");
```

Comment:
Can it be too late to set it here? E.g. smth in SYCL RT already acted on its value

Reply:
I think we should ask users of SYCL_USM_HOSTPTR_IMPORT=1 to also set SYCL_HOST_UNIFIED_MEMORY=1 in the doc.
This code can stay however.